stl/inc/deque

```diff
@@ -1237,37 +1255,25 @@ public:
 
         if (_UFirst != _ULast) {
             if (_Off <= _Mysize() / 2) { // closer to front, push to front then rotate
-                _TRY_BEGIN
+                _Restore_old_size_guard<true> _Guard{this, _Oldsize};

```

Comment:
Ditto boolean arguments are code smells. 

I have no Idea what true means here without looking at the code `_Restore_old_size_guard<true>` 

If we would go with `_Restore_old_size_guard<_Direction::_Front>`, it would be obvious

Reply:
What I don't like in this enumerating bools, is that you can't just call it `_Direction`, as it is too generic. For previous case it was even namespace scope. When called properly, lines become long.